src/anyio/_core/_sockets.py

```diff
@@ -245,7 +247,9 @@ async def try_connect(remote_host: str, event: Event) -> None:
     return connected_stream
 
 
-async def connect_unix(path: str | PathLike[str]) -> UNIXSocketStream:
+async def connect_unix(
+    path: str | bytes | PathLike[str] | PathLike[bytes]
```

Comment:
Hm, wouldn't `PathLike[Any]` work here, given that the variable is bound to `str` and `bytes` anyway?

Reply:
Pylance gave me an error when I first tried, but Pyright doesn't. Not sure what's going on there.. I'll trust the CLI and change it.

Reply:
If the CI passes, then that's good enough for me.